tools/hangar/__snapshots__/test_corpus/valid/inflight_class_capture_preflight_object.test.w_compile_tf-aws.md

```diff
@@ -40,3 +546,245 @@ $APP.synth();
 //# sourceMappingURL=preflight.cjs.map
 ```
 
+## preflight.file1-3.cjs
+```cjs
+"use strict";
+const $stdlib = require('@winglang/sdk');
+const std = $stdlib.std;
+const $helpers = $stdlib.helpers;
+const $extern = $helpers.createExternRequire(__dirname);
+let $preflightTypesMap = {};
+const blah = require("./preflight.inner-2.cjs");
+Object.assign($preflightTypesMap, blah.$preflightTypesMap);
+const cloud = $stdlib.cloud;
+const util = $stdlib.util;
+class Foo extends $stdlib.std.Resource {
+  constructor($scope, $id, ) {
+    super($scope, $id);
+  }
+  foo() {
+    return "foo";
+  }
+  checkWidget(widget) {
+    return ((widget.compute()) + (blah.Widget.staticCompute(this)));
+  }
+  static _toInflightType() {
+    return `
+      require("${$helpers.normalPath(__dirname)}/inflight.Foo-2.cjs")({
+      })
+    `;
+  }
+  _toInflight() {
+    return `
+      (await (async () => {
+        const FooClient = ${Foo._toInflightType()};
+        const client = new FooClient({
+        });
+        if (client.$inflight_init) { await client.$inflight_init(); }
+        return client;
+      })())
+    `;
+  }
+  get _liftMap() {
+    return ({
+      "$inflight_init": [
+      ],
+    });
+  }
+}
+module.exports = { $preflightTypesMap, Foo };
```

Comment:
I'm aiming to get a single `$preflightTypesMap` with **all** the (inflight) classes defined preflight. To achieve this each local `$preflightTypesMap` aggregates the inner ``$preflightTypesMap`s it imports. This doesn't violate wing's export rules because this map isn't being used by application code. It's only being used in the lift mpas, where we need access to all types. Note that `$preflightTypesMap` is indexed by a unique id per type, so there shouldn't be any naming conflicts here. The top comment to this PR tries to explain the mechanics of this.